### PR TITLE
GDB-3947 Introduce plugin registry and tests

### PR DESCRIPTION
**Pull request by svilenvelikov**

### CODE REVIEW

File: webpack.config.dev.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = merge(commonConfig, {
     devtool: 'source-map',
     // devtool: 'none',
     output: {
-        filename: '[name].js',
+        filename: '[name].bundle.js',
```
[the remark this answers is not in the document]
Comment (svilenvelikov) [reply webpack.config.dev.js]: Maybe I made it by mistake. Thx.

File: webpack.config.common.js [hunk 2]
```diff
@@ -14,7 +20,18 @@ module.exports = {
...
+        new MergeIntoSingleFilePlugin({
+            files: {
+                "plugins.js": [
+                    'src/**/plugin.js'
+                ]
+            }
+        }),
         new CopyPlugin([
+            {
+                from: 'src/js/angular/plugin-registry.js',
```
[the remark this answers is not in the document]
Comment (svilenvelikov) [reply webpack.config.common.js]: I need the PluginRegistry loaded and present in the global window scope before the `plugins.js` gets loaded and run as it needs the registry in order to add the available plugins.

File: test/plugin-registry.spec.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        registry.add('route', {
+            'order': 1,
+            'stateName': 'login'
+        });
+
+        const result = registry.get('route');
+
+        expect(result.length).toEqual(2);
+
+        expect(result[0].order).toEqual(1);
```
[the remark this answers is not in the document]
Comment (svilenvelikov) [reply test/plugin-registry.spec.js]: done

File: test/plugin-registry.spec.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            'stateName': 'search'
+        });
+
+        registry.add('route', {
+            'order': 2,
+            'priority': 2,
+            'stateName': 'login'
+        });
+
+        expect(registry.get('route').length).toEqual(1);
```
[the remark this answers is not in the document]
Comment (svilenvelikov) [reply test/plugin-registry.spec.js]: done

File: webpack.config.dev.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (svilenvelikov) [reply webpack.config.dev.js]: removed

File: webpack.config.common.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mihailradkov) [reply webpack.config.common.js]: I see. OK

**Pull request merged**